assets/chat/js/menus.js

```diff
@@ -168,31 +169,41 @@ class ChatSettingsMenu extends ChatMenu {
     }
 
     updateNotification(){
-        const perm = Notification.permission === 'default' ? 'required' : Notification.permission
-        this.notificationEl.text(`(Permission ${perm})`)
+        if( Notification.permission === "granted" ) {
+            this.enableNotificationsBtn.text("Permission granted ✓");
+            this.enableNotificationsBtn.attr('disabled', true);
+            this.notificationsFieldset.attr('disabled', false);
+        } else {
+            this.enableNotificationsBtn.text("Enable Notifications");
+            this.enableNotificationsBtn.attr('disabled', false);
+            this.notificationsFieldset.attr('disabled', true);
+        }
     }
 
     notificationPermission(){
         return new Promise((resolve, reject) => {
-            switch(Notification.permission) {
-                case 'default':
-                    Notification.requestPermission(permission => {
-                        switch(permission) {
-                            case 'granted':
-                                resolve(permission);
-                                break;
-                            default:
-                                reject(permission);
-                        }
-                    });
-                    break;
-                case 'granted':
-                    resolve(Notification.permission);
-                    break;
-                case 'denied':
-                default:
-                    reject(Notification.permission);
-                    break;
+            // Detecting for Blink webengine and whether we are not the top frame
+            // See issue github.com/MemeLabs/chat-gui/issues/12
+            const isBlink = window.Intl && Intl.v8BreakIterator;
```

Comment:
I think `Intl` will eventually be supported by all browsers, and `Intl.v8BreakIterator` is unofficial and slated for removal. We will need to find another way to detect Blink.

Reply:
I see. 
I think the next best thing would be a user-agent library with a whitelist.
So the custom popup would be the default action, and browsers on the whitelist would get a direct permission request.